tests/Query/AggregationBuilderTest.php

```diff
@@ -0,0 +1,147 @@
+use MongoDB\Laravel\Query\AggregationBuilder;
+use MongoDB\Laravel\Tests\Models\User;
+use MongoDB\Laravel\Tests\TestCase;
+
+class AggregationBuilderTest extends TestCase
+{
+    public function tearDown(): void
+    {
+        User::truncate();
+
+        parent::tearDown();
+    }
+
+    public function testCreateAggregationBuilder(): void
+    {
+        User::insert([
+            ['name' => 'John Doe', 'birthday' => new UTCDateTime(new DateTimeImmutable('1989-01-01'))],
+            ['name' => 'Jane Doe', 'birthday' => new UTCDateTime(new DateTimeImmutable('1990-01-01'))],
+        ]);
+
+        // Create the aggregation pipeline from the query builder
+        $pipeline = User::aggregate();
+
+        $this->assertInstanceOf(AggregationBuilder::class, $pipeline);
+
+        $pipeline
+            ->match(name: 'John Doe')
+            ->limit(10)
+            ->addFields(
+                // Requires MongoDB 5.0+
+                year: Expression::year(
+                    Expression::dateFieldPath('birthday'),
+                ),
+            )
+            ->sort(year: Sort::Desc, name: Sort::Asc)
+            ->unset('birthday');
+
+        // Compare with the expected pipeline
+        $expected = [
+            ['$match' => ['name' => 'John Doe']],
+            ['$limit' => 10],
+            [
+                '$addFields' => [
+                    'year' => ['$year' => ['date' => '$birthday']],
+                ],
+            ],
+            ['$sort' => ['year' => -1, 'name' => 1]],
+            ['$unset' => ['birthday']],
+        ];
+
+        $this->assertSamePipeline($expected, $pipeline->getPipeline());
+
+        // Execute the pipeline and validate the results
+        $results = $pipeline->get();
+        $this->assertInstanceOf(Collection::class, $results);
+        $this->assertCount(1, $results);
+        $this->assertInstanceOf(ObjectId::class, $results->first()['_id']);
+        $this->assertSame('John Doe', $results->first()['name']);
+        $this->assertIsInt($results->first()['year']);
+        $this->assertArrayNotHasKey('birthday', $results->first());
+
+        // Execute the pipeline and validate the results in a lazy collection
+        $results = $pipeline->cursor();
+        $this->assertInstanceOf(LazyCollection::class, $results);
+
+        // Execute the pipeline and return the first result
+        $result = $pipeline->first();
+        $this->assertIsArray($result);
+        $this->assertInstanceOf(ObjectId::class, $result['_id']);
+        $this->assertSame('John Doe', $result['name']);
+    }
+
+    public function testAddRawStage(): void
+    {
+        $collection = $this->createMock(MongoDBCollection::class);
+
+        $pipeline = new AggregationBuilder($collection);
+        $pipeline
+            ->addRawStage('$match', ['name' => 'John Doe'])
+            ->addRawStage('$limit', 10)
+            ->addRawStage('$replaceRoot', (object) ['newRoot' => '$$ROOT']);
```

Comment:
Is there any reason to perform an object cast here?

I noted that `assertSamePipeline()` normalizes the actual and expected values before comparison, but this seems more like a test of _that_ behavior than anything to do with `addRawStage()`.

Reply:
This tests that `addRawStage` accepts `stdClass` as 2nd argument.